### DISCUSSION TITLE
How to unpack list parser to another list parser

### DISCUSSION THREAD

**Discussion by ykmnkmi** [petitparser/dart-petitparser#76] [Q&A] [answered]
```dart
Parser<Token> expressionStart = ...;
Parser<Token> expressionEnd = ...;
Parser<List<Token>> expressionBody = ...;
Parser<List<Object>> expression = expressionStart & expressionBody & expressionEnd;
```
and unpacking expression value to flat list

some thing like ` expressionStart & ...expressionBody & expressionEnd` will be good
```dart
Parser<List<Token>> expression = expressionStart & expressionBody.unpack() & expressionEnd;
```

**Comment by renggli** [comment]
Not sure what you expect `unpack()` to do, but there is [pick](https://github.com/petitparser/dart-petitparser/blob/master/petitparser/lib/src/parser/action/pick.dart) that returns the n-th element of a list:

```dart
Parser<List<Token>> expression = (expressionStart & expressionBody & expressionEnd)
    .pick(1);
```

**Comment by ykmnkmi** [comment]
my task is getting flatten list of tokens `Parser<List<Token>>` expression.
`expressionStart` and `expressionEnd` needed too.
`Parser<List<Token>> expressionBody = ...;` list of token
and i can't assign
`Parser<List<Token>> expression = expressionStart & expressionBody & expressionEnd`

(sorry for my poor english ☹)

**Comment by renggli** [comment, marked as answer]
This also includes the start and end tokens:

```dart
Parser<List<Token>> expression = (expressionStart & expressionBody & expressionEnd)
    .map((parts) => [parts[0], ...parts[1], parts[2]]);
```
